wordpress-mobile/WordPress-iOS
# Remove Site Name experiment code and set feature flag to disabled

Fixes #18883

### Description

This removes the experiment code (A/B test) for the Site Name feature, and disables the feature flag. It also removes the associated tracking events, which are no longer useful.

### To test:

1. Start the site creation flow
2. Navigate through the experience to the last step (choosing a domain)
3. Ensure that the site name screen is not displayed and that the choose a domain screen _is_ displayed

## Regression Notes
1. Potential unintended areas of impact
Site creation steps

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
